.github/workflows/lint.yml

```diff
@@ -0,0 +1,28 @@
+# Lint tests run on PR
+# but should not run after push to main because reporting
+# these after push is meaningless to the building of the package
+name: Lint
+
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  lint:
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
I think that for pure Python linting, we can run only on ubuntu-latest, this should save some runtime & reduce visual pollution in the number of checks